packages/coverage-istanbul/src/provider.ts

```diff
@@ -204,3 +218,24 @@ async function transformCoverage(coverageMap: CoverageMap) {
 function removeQueryParameters(filename: string) {
   return filename.split('?')[0]
 }
+
+function resolveYarnPnpPaths(coverage: CoverageMapData) {
+  if (!pnp) {
+    return
+  }
+
+  for (const file of Object.keys(coverage) as (keyof typeof coverage)[]) {
+    const fileCoverage = coverage[file]
+
+    const filename = pnp.resolveVirtual(file)
```

Comment:
This should only be called on virtual paths, otherwise it will return `null`.

Reply:
Also, on windows this will return a `\` separated path, where the coverage map contains `/` separated paths (even on windows), so you'll need to apply your favourite `slash` function (or just `.replaceAll('\\', '/')`)